core/src/main/java/de/danoeh/antennapod/core/util/download/ConnectionStateMonitor.java

```diff
@@ -0,0 +1,47 @@
+package de.danoeh.antennapod.core.util.download;
+
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.NetworkCapabilities;
+import android.net.NetworkRequest;
+import android.os.Build;
+import android.util.Log;
+
+import androidx.annotation.RequiresApi;
+import de.danoeh.antennapod.core.util.NetworkUtils;
+
+@RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+public class ConnectionStateMonitor
+        extends ConnectivityManager.NetworkCallback
+        implements ConnectivityManager.OnNetworkActiveListener {
+    private static final String TAG = "ConnectionStateMonitor";
+    final NetworkRequest networkRequest;
+
+    public ConnectionStateMonitor() {
+        networkRequest = new NetworkRequest.Builder()
+                .addTransportType(NetworkCapabilities.TRANSPORT_CELLULAR)
+                .addTransportType(NetworkCapabilities.TRANSPORT_WIFI)
+                .build();
+    }
+
+    public void onNetworkActive() {
```

Comment:
This is the interface method, right? Could you add the override annotation to make that more clear, please?